REST-API-project/book-api.js

```diff
@@ -0,0 +1,102 @@
+//importing the express framework
+const express = require('express')
+const bodyParser = require('body-parser');
+const cors = require('cors');
+
+//creating the express app
+const app = express();
+//setting the port
+const port = process.env.PORT || 3000;
+
+// Where we will keep books
+let books = [];
+
+app.use(cors());
+
+// Configuring body parser middleware
+app.use(bodyParser.urlencoded({ extended: false }));
+app.use(bodyParser.json());
+
+//what you get on the main page localhost:3000
+app.get('/', (req, res) => {
+    res.send('Hello World, from Express');
+  });
+
+  //adding books to the book list
+app.post('/book', (req, res) => {
+    const book = req.body;
+
+    // Output the book to the console for debugging
+    console.log(book);
+    books.push(book);
+
+    res.send('Book is added to the database');
+
+});
+
+
+app.get('/newbook', (req, res) => {
+    res.sendFile(`${__dirname}/new-book.html`);
+});
+
+//getting the book list
+app.get('/booklist', (req, res) => {
+    res.sendFile(`${__dirname}/book-list.html`);
+});
+
+app.get('/books', (req, res) => {
+    res.json(books);
+});
+
+
+//Retrieving a Book by ISBN
+app.get('/book/:isbn', (req, res) => {
+    // Reading isbn from the URL
+    const isbn = req.params.isbn;
+
+    // Searching books for the isbn
+    for (let book of books) {
+        if (book.isbn === isbn) {
+            res.json(book);
+            return;
+        }
+    }
+
+    // Sending 404 when not found something is a good practice
+    res.status(404).send('Book not found');
```

Comment:
Suggestion: Instead of writing of just saying 'Book not found', you can say 'Book is not found'